tree-sitter require std=c99

tree-sitter require at least std=c99 to be built without errors.
Not all compilers use the std c99 as default and such special care should be made when using such features.

Actual output:
```
tree-sitter/lib/src/././language.h:67:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
     for (unsigned i = 0; i < section_count; i++) {
     ^
[...]
```

Here is a quick fix:
``` diff
diff --git a/shlr/Makefile b/shlr/Makefile
index 0c10f4379..0946b24fd 100644
--- a/shlr/Makefile
+++ b/shlr/Makefile
@@ -336,7 +336,7 @@ tree-sitter/libtree-sitter.$(EXT_AR): tree-sitter/lib/src/lib.o
        $(RANLIB) $@

 tree-sitter/lib/src/lib.o: tree-sitter-sync
-       $(CC) -c tree-sitter/lib/src/lib.c -o $@ -Itree-sitter/lib/include -Itree-sitter/lib/src $(CFLAGS)
+       $(CC) -std=c99 -c tree-sitter/lib/src/lib.c -o $@ -Itree-sitter/lib/include -Itree-sitter/lib/src $(CFLAGS)
```

--- a/shlr/Makefile
+++ b/shlr/Makefile
@@ -336,7 +336,7 @@ tree-sitter/libtree-sitter.$(EXT_AR): tree-sitter/lib/src/lib.o
 	$(RANLIB) $@
 
 tree-sitter/lib/src/lib.o: tree-sitter-sync
-	$(CC) std=c99 -c tree-sitter/lib/src/lib.c -o $@ -Itree-sitter/lib/include -Itree-sitter/lib/src $(CFLAGS)
+	$(CC) -std=c99 -c tree-sitter/lib/src/lib.c -o $@ -Itree-sitter/lib/include -Itree-sitter/lib/src $(CFLAGS)
 
 tree-sitter-sync:
 	"$(SHELL)" clone_3rd_repo.sh tree-sitter "${TS_URL}" "${TS_BRA}" "${TS_TIP}"

--- a/shlr/Makefile
+++ b/shlr/Makefile
@@ -29,7 +29,7 @@ endif
 # NOTE: when you update TS_TIP or TS_BRA, also update them in shlr/meson.build
 TS_URL=https://github.com/tree-sitter/tree-sitter.git
 TS_BRA=master
-TS_TIP=80008b0bccbddffc8e68f66a5f173ef71fd125e3
+TS_TIP=9f77afcd1ff050855259f83fcf50056ad842a790
 
 # NOTE: when you update SHELLPARSER_TIP or SHELLPARSER_BRA, also update them in shlr/meson.build
 SHELLPARSER_URL=https://github.com/ret2libc/radare2-shell-parser.git
@@ -336,7 +336,7 @@ tree-sitter/libtree-sitter.$(EXT_AR): tree-sitter/lib/src/lib.o
 	$(RANLIB) $@
 
 tree-sitter/lib/src/lib.o: tree-sitter-sync
-	$(CC) -c tree-sitter/lib/src/lib.c -o $@ -Itree-sitter/lib/include -Itree-sitter/lib/src $(CFLAGS)
+	$(CC) std=c99 -c tree-sitter/lib/src/lib.c -o $@ -Itree-sitter/lib/include -Itree-sitter/lib/src $(CFLAGS)
 
 tree-sitter-sync:
 	"$(SHELL)" clone_3rd_repo.sh tree-sitter "${TS_URL}" "${TS_BRA}" "${TS_TIP}"

--- a/shlr/meson.build
+++ b/shlr/meson.build
@@ -246,7 +246,7 @@ if res.returncode() == 0
   endif
 
   # NOTE: when you update TS_TIP or TS_BRA, also update them in shlr/Makefile
-  TS_TIP = '80008b0bccbddffc8e68f66a5f173ef71fd125e3'
+  TS_TIP = '9f77afcd1ff050855259f83fcf50056ad842a790'
   TS_BRA = 'master'
 
   message('Cloning tree-sitter ' + TS_BRA + ' branch, commit ' + TS_TIP + ', into ' + tree_sitter_path)
@@ -271,7 +271,8 @@ tree_sitter_inc = [platform_inc, include_directories('tree-sitter/lib/src'), inc
 
 libtree_sitter = static_library('tree_sitter', tree_sitter_files,
   include_directories: tree_sitter_inc,
-  implicit_include_directories: false
+  implicit_include_directories: false,
+  c_args: ['-std=c99']
 )
 
 tree_sitter_dep = declare_dependency(

